pkg/catalog/catalog.go

```diff
@@ -853,27 +857,27 @@ func (c *Catalog) ListCommits(ctx context.Context, repository string, branch str
 	}); err != nil {
 		return nil, false, err
 	}
-	branchCommitID, err := c.Store.Dereference(ctx, repositoryID, graveler.Ref(branchRef))
+	resolvedRef, err := c.Store.Dereference(ctx, repositoryID, graveler.Ref(branchRef))
 	if err != nil {
 		return nil, false, fmt.Errorf("branch ref: %w", err)
 	}
-	if branchCommitID == "" {
+	if resolvedRef.CommitID == "" {
 		// return empty log if there is no commit on branch yet
 		return make([]*CommitLog, 0), false, nil
 	}
```

Comment:
I see this pattern returns several times, maybe have a utility function to handle missing deref + checking for commit.

Reply:
true, the dereference was just to a commit